frontend/src/router.js

```diff
@@ -600,27 +609,50 @@ async function requireCamp(to, from, next) {
   }
 }
 
-async function requireScheduleEntry(to, from, next) {
-  const scheduleEntry = await scheduleEntryFromRoute(to)
-  if (scheduleEntry === undefined) {
-    next({
-      name: 'PageNotFound',
-      params: [to.fullPath, ''],
-      replace: true,
-    })
-  } else {
-    await scheduleEntry._meta.load
-      .then(() => {
+async function requireActivityScheduleEntry(to, from, next) {
+  await apiStore
+    .get()
+    .activities({ id: to.params.activityId })
+    ._meta.load.then((activity) => {
+      // activity exists
+      if (to.params.scheduleEntryId) {
+        apiStore
+          .get()
+          .scheduleEntries({ id: to.params.scheduleEntryId })
+          ._meta.load.then(() => {
```

Comment:
minor: i find it a little silly that you mix the await syntax with then

Reply:
I find it perfectly valid in use cases where writing it with only await would be more complicated, especially when you have to use `then` and `catch` at the same time (which would translate to `await` with a `try {} catch {}` block). In particular, I think `then` is more useful and readable when you have to continue working with a promise (e.g. when chaining multiple transformations or steps), while `await` is useful when you are only interested in the result payload of the promise.
Regardless of code style preferences, we have this mixed syntax already in various places in the frontend as well as in hal-json-vuex.